### DISCUSSION TITLE
How to use type specs?

### DISCUSSION THREAD

**Discussion by akasprzok** [elixir-protobuf/protobuf#294] [Q&A]
Seems that typespecs are created via a before_compile callback [here](https://github.com/elixir-protobuf/protobuf/blob/4814ec035ea828d3f0932cc978bf05f54e85d0e1/lib/protobuf/dsl.ex#L140)

I've got a handful of compiled Protobufs in a .pb.ex file, including

```elixir
defmodule Logproto.PushRequest do
  @moduledoc false
  use Protobuf, protoc_gen_elixir_version: "0.10.0", syntax: :proto3

  field(:streams, 1, repeated: true, type: Logproto.StreamAdapter, deprecated: false)
end
```

But whenever I run `mix docs`, using Logproto.PushRequest.t() in a typespec, I got warnings like 

```
warning: documentation references type "Logproto.PushRequest.t()" but it is hidden or private
```

Will dig into this more, but is there something obvious I might be missing!

**Comment by whatyouhide** [comment]
Are you writing `Logproto.PushRequest.t/0` or `Logproto.PushRequest.t()` in a documentation string?

**Reply by akasprzok** [reply]
For example:

```elixir
@type timestamp :: Google.Protobuf.Timestamp.t()
@type entry: Logproto.EntryAdapter.t()
```

```
mix docs
...
warning: documentation references function "Google.Protobuf.Timestamp.new/0" but it is hidden
  lib/sleipnir/timestamp.ex:8: Sleipnir.Timestamp.new/0

warning: documentation references type "Logproto.EntryAdapter.t()" but it is hidden or private
  lib/sleipnir.ex:23: t:Sleipnir.entry/0
  ```
  
  Can be repro'd using [my little Loki client project](https://github.com/akasprzok/sleipnir/tree/main/), just run `mix docs` on `main`.

**Reply by whatyouhide** [reply]
I see. Opening an issue for this, it shouldn't warn.
